app/server/methods/activities.js

```diff
@@ -335,25 +339,30 @@ export default Meteor.methods({
     // aggregate activities into daily bins grouped by type
     //  - activity count
     //  - activity minutes
-    return d3
-      .nest()
-      .key(function (activity) {
+    const groupedData = d3.groups(
+      annotatedActivities,
+      function (activity) {
         return activity.activityTypeName;
-      })
-      .key(function (activity) {
+      },
+      function (activity) {
         return moment(activity.activityDate).startOf('day').toDate();
-      })
-      .rollup(function (dailyActivities) {
-        return {
-          activity_count: dailyActivities.length,
-          activity_minutes: d3.sum(dailyActivities, function (
-            activity
-          ) {
-            return parseFloat(activity.duration);
-          }),
-        };
-      })
-      .entries(annotatedActivities);
+      }
+    );
+
+    return d3.rollup(groupedData, function (dailyActivities) {
+      return {
+        key: dailyActivities[0],
+        values: dailyActivities[1].map((activity) => ({
```

Comment:
I suspect this intermediary object would be an `activityType`, but it 8s not clear from the code. How can we improve the readability of these grouping and roll up functions? E.g. can we use named properties instead of array indices?

Reply:
I will add comments. and intermediate names... the key names final object are being used in charts in the client so those names were kept the same.

Reply:
done